Cabal/Distribution/Backpack/UnifyM.hs

```diff
@@ -108,31 +139,67 @@ instance Applicative (UnifyM s) where
     UnifyM f <*> UnifyM x = UnifyM $ \r -> do
         f' <- f r
         case f' of
-          Left err -> return (Left err)
-          Right f'' -> do
+          Nothing -> return Nothing
+          Just f'' -> do
               x' <- x r
               case x' of
-                  Left err -> return (Left err)
-                  Right x'' -> return (Right (f'' x''))
+                  Nothing -> return Nothing
+                  Just x'' -> return (Just (f'' x''))
```

Comment:
Unfortunately not, because it's IO (Maybe a) so an fmap will use the IO instance, not the MaybeT IO instance. But we can't depend on MaybeT, so it's done by hand...